### PR TITLE
Add a `testUtils.createQueue` test helper function

### PR DESCRIPTION
**Pull request by yjpa7145**

### CODE REVIEW

File: cumulus/tasks/queue-granules/tests/index.js [hunk 1]
```diff
@@ -14,16 +14,16 @@ const workflowTemplate = require('./fixtures/workflow-template.json');
...
+  t.context.queueUrl = await createQueue();
   await s3().createBucket({ Bucket: t.context.bucket }).promise();
 });
 
-test.afterEach.always(async (t) => {
-  await recursivelyDeleteS3Bucket(t.context.bucket);
-  await sqs().deleteQueue({ QueueUrl: `http://${process.env.LOCALSTACK_HOST}:4576/queue/${t.context.queue}` }).promise();
-});
+test.afterEach.always(async (t) =>
+  Promise.all([
```
Comment (scisco) [inline cumulus/tasks/queue-granules/tests/index.js]: I think the `await` is missing in front of `Promise.all`.

File: cumulus/tasks/queue-pdrs/tests/index.js [hunk 2]
```diff
@@ -14,15 +14,17 @@ const workflowTemplate = require('./fixtures/workflow-template.json');
 const aws = require('@cumulus/common/aws');
 
 test.beforeEach(async (t) => {
-  t.context.bucket = testUtils.randomString();
-  await sqs().createQueue({ QueueName: 'testQueue' }).promise();
+  t.context.queueUrl = await createQueue();
```
Comment (scisco) [inline cumulus/tasks/queue-pdrs/tests/index.js]: This is very neat. 💯 d

File: cumulus/tasks/queue-pdrs/tests/index.js [hunk 3]
```diff
@@ -14,15 +14,17 @@ const workflowTemplate = require('./fixtures/workflow-template.json');
...
+  t.context.bucket = randomString();
   return s3().createBucket({ Bucket: t.context.bucket }).promise();
 });
 
-test.afterEach.always(async (t) => {
-  await recursivelyDeleteS3Bucket(t.context.bucket);
-  await sqs().deleteQueue({ QueueUrl: `http://${process.env.LOCALSTACK_HOST}:4576/queue/testQueue` }).promise();
-});
+test.afterEach.always((t) =>
+  Promise.all([
```
Comment (scisco) [inline cumulus/tasks/queue-pdrs/tests/index.js]: I think we should use async/await here too, unless, we don't really need to wait for this finish.

File: packages/common/test-utils.js [hunk 4]
```diff
@@ -84,3 +86,24 @@ function testAwsClient(Service, options) {
...
+async function createQueue() {
+  const createQueueResponse = await aws.sqs().createQueue({
+    QueueName: exports.randomString()
+  }).promise();
+
+  // Properly set the Queue URL.  This is needed because LocalStack always
+  // returns the QueueUrl as "localhost", even if that is not where it should
+  // actually be found.  CircleCI breaks without this.
+  const returnedQueueUrl = url.parse(createQueueResponse.QueueUrl);
+  returnedQueueUrl.host = undefined;
```
Comment (scisco) [inline packages/common/test-utils.js]: is this to make it http/https agnostic?

**Comment by jennyhliu** [comment]
It looks good to me.  Make sure the deleted function createQueue under packages/ingest/aws.js is not used.

File: cumulus/tasks/queue-granules/tests/index.js (hunk 1 above)
Comment (yjpa7145) [reply cumulus/tasks/queue-granules/tests/index.js]: Actually, the async is not needed.  Ava will handle the Promise that's being returned by Promise.all.

File: cumulus/tasks/queue-pdrs/tests/index.js (hunk 3 above)
Comment (yjpa7145) [reply cumulus/tasks/queue-pdrs/tests/index.js]: Similar to what I added in my previous comment, ava will wait for the Promise returned by Promise.all to resolve.

File: packages/common/test-utils.js (hunk 4 above)
Comment (yjpa7145) [reply packages/common/test-utils.js]: This has to do with the way that the `url.format` function determines what host to return.

https://nodejs.org/docs/v6.10.3/api/url.html#url_url_format_urlobject

**Pull request merged**